src/views/virtualmachines/list/components/VirtualMachineRow/components/CPUPercentage.tsx

```diff
@@ -1,22 +1,41 @@
 import React, { FC } from 'react';
 
-import { V1CPU, V1VirtualMachine } from '@kubevirt-ui-ext/kubevirt-api/kubevirt';
+import { V1VirtualMachine, V1VirtualMachineInstance } from '@kubevirt-ui-ext/kubevirt-api/kubevirt';
+import { VMQueries } from '@kubevirt-utils/components/Charts/utils/queries';
+import { getCPU, getVCPUCount } from '@kubevirt-utils/resources/vm';
 import { NO_DATA_DASH } from '@kubevirt-utils/resources/vm/utils/constants';
 import { isEmpty } from '@kubevirt-utils/utils/utils';
-import { getCPUUsagePercentage } from '@virtualmachines/list/metrics';
+import { PrometheusEndpoint, usePrometheusPoll } from '@openshift-console/dynamic-plugin-sdk';
+import useDuration from '@virtualmachines/details/tabs/metrics/hooks/useDuration';
 import { isRunning } from '@virtualmachines/utils';
 
 type CPUPercentageProps = {
+  queries: Record<VMQueries, string>;
   vm: V1VirtualMachine;
-  vmiCPU: V1CPU;
+  vmi?: V1VirtualMachineInstance;
 };
 
-const CPUPercentage: FC<CPUPercentageProps> = ({ vm, vmiCPU }) => {
-  const cpuUsagePercentage = getCPUUsagePercentage(vm, vmiCPU);
+const CPUPercentage: FC<CPUPercentageProps> = ({ queries, vm, vmi }) => {
+  const { currentTime } = useDuration();
 
-  if (isEmpty(cpuUsagePercentage) || !isRunning(vm)) return <span>{NO_DATA_DASH}</span>;
+  const [dataCPUUsage] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint?.QUERY,
+    endTime: currentTime,
+    namespace: vmi?.metadata?.namespace,
+    query: queries?.CPU_USAGE,
+  });
+  const vmCPU = getCPU(vmi);
 
-  return <span>{cpuUsagePercentage.toFixed(2)}%</span>;
+  const cpuUsage = +(dataCPUUsage?.data?.result?.[0]?.value?.[1] || 0);
+  const cpuRequested = getVCPUCount(vmCPU);
+
+  const averageCPUUsageStr =
+    cpuRequested !== 0 ? ((cpuUsage / cpuRequested) * 100).toFixed(2) || 0 : 0;
+  const averageCPUUsage = Number(averageCPUUsageStr);
+
+  if (isEmpty(dataCPUUsage) || !isRunning(vm)) return <span>{NO_DATA_DASH}</span>;
```

Comment:
I would move this line before the constant declaration since if this condition is true than we don't need to declare them